src/frontend/src/hooks/finance.hooks.ts

```diff
@@ -369,9 +369,12 @@ export const useCreateAccountCode = () => {
  */
 export const useCreateVendor = () => {
   const queryClient = useQueryClient();
-  return useMutation<{ message: string }, Error, any>(['vendors', 'create'], async (vendorData: { name: string }) => {
-    const { data } = await createVendor(vendorData);
-    queryClient.invalidateQueries(['vendors']);
-    return data;
-  });
+  return useMutation<{ message: string }, Error, { name: string }>(
```

Comment:
the first <{message: string}> should actually be a Vendor, the code here before was technically wrong